### PR TITLE
overlord,daemon: adjust startup timeout via EXTEND_TIMEOUT_USEC using an estimate

### PR DESCRIPTION
**Pull request by pedronis**
This is based on having now separated out the expensive parts as Overlord.StartUp (#7132).

Come with a spread test, though that requires some dedicated sleep code :/

TODO: adjust estimate calculation params

We could make this dynamic over StartUp execution by measuring
effective time / snaps processed, that would add quite a bit of
complexity though.

### CODE REVIEW

**Comment by chipaca** [comment]
I think we do want this, fwiw

**Comment by pedronis** [comment]
this seems to have failed for real on 16.04, need to dig, moving it to 2.42

**Comment by pedronis** [comment]
ah, EXTEND_TIMEOUT_USEC was only in 236, https://github.com/systemd/systemd/blob/master/NEWS

:(

**Comment by zyga** [comment]
@pedronis can we still tell systemd abut and just not test it on older systems or is this a complete no-go without prior systemctl version query?

File: tests/main/snapd-slow-startup/task.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    cat > /etc/systemd/system/snapd.service.d/slow-startup.conf <<EOF
+    [Service]
+    Environment=SNAPD_SLOW_STARTUP=55s
+    Restart=no
+    TimeoutStartSec=50s
+    EOF
+    systemctl daemon-reload
+
+    # startup timeout will be adjusted by 30s + 7 * 5s and this will succeed
+    systemctl start snapd.service snapd.socket
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply tests/main/snapd-slow-startup/task.yaml]: No luck there, apparently systemd is not updating TimeoutStartUSec. I think we can land it as it is.

**Pull request merged**